p2p/sync/sync.go

```diff
@@ -318,7 +318,10 @@ func (s *BlockFetcher) adaptAndSanityCheckBlock(
 
 			var coreTxs []core.Transaction
 			for _, tx := range txs {
-				coreTxs = append(coreTxs, p2p2core.AdaptTransaction(tx, s.network))
+				coreTx, err := p2p2core.AdaptTransaction(tx, s.network)
+				if err == nil {
+					coreTxs = append(coreTxs, coreTx)
+				}
```

Comment:
We should fail fast, because missing a transaction results in a completely different and invalid block.